tests/EdFi.Admin.DataAccess.IntegrationTests/Repositories/PostgreSQL/ClientAppRepoTests.cs

```diff
@@ -0,0 +1,727 @@
+        public void Setup()
+        {
+            // Read settings
+            AppContext.SetSwitch("Npgsql.EnableLegacyTimestampBehavior", true);
+
+            var builder = new ConfigurationBuilder()
+               .AddJsonFile($"appSettings.json", true, true)
+               .AddJsonFile($"appSettings.development.json", true, true);
+
+            var config = builder.Build();
+
+            // Setup PostgreSQL
+            var connectionString = config.GetConnectionString("PostgreSQL");
+
+            var optionsBuilder = new DbContextOptionsBuilder();
+            optionsBuilder.UseNpgsql(connectionString);
+            Context = new PostgresUsersContext(optionsBuilder.Options);
+            OpenContext = new PostgresUsersContext(optionsBuilder.Options);
+
+            // Startup a transaction so we can dispose of any changes after running the tests
+            Transaction = new TransactionScope();
+
+            // Configure settings mocks
+            var usersContextFactory = A.Fake<IUsersContextFactory>();
+            A.CallTo(() => usersContextFactory.CreateContext())
+                .Returns(Context);
+
+            var configValueProviderStub = A.Fake<IConfigurationRoot>();
+
+            var tokenTimeoutSection = A.Fake<IConfigurationSection>();
+            A.CallTo(() => tokenTimeoutSection.Value).Returns(BearerTokenTimeoutMinutes.ToString());
+            A.CallTo(() => configValueProviderStub.GetSection("BearerTokenTimeoutMinutes")).Returns(tokenTimeoutSection);
+
+            var contextSection = A.Fake<IConfigurationSection>();
+            A.CallTo(() => contextSection.Value).Returns(DefaultOperationalContextUri);
+            A.CallTo(() => configValueProviderStub.GetSection("DefaultOperationalContextUri")).Returns(contextSection);
+
+            var appNameSection = A.Fake<IConfigurationSection>();
+            A.CallTo(() => appNameSection.Value).Returns(DefaultApplicationName);
+            A.CallTo(() => configValueProviderStub.GetSection("DefaultApplicationName")).Returns(appNameSection);
+
+            var claimSetSection = A.Fake<IConfigurationSection>();
+            A.CallTo(() => claimSetSection.Value).Returns(DefaultClaimSetName);
+            A.CallTo(() => configValueProviderStub.GetSection("DefaultClaimSetName")).Returns(claimSetSection);
+
+            // Create the system under test
+            _clientAppRepo = new ClientAppRepo(usersContextFactory, configValueProviderStub);
+
+            // Load a fake API client
+            _testClient = new ApiClient(true)
+            {
+                Name = "ClientAppRepoTest" + Guid.NewGuid().ToString("N")
+            };
+            Context.Clients.Add(_testClient);
+            Context.SaveChangesForTest();
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            Transaction.Dispose();
+        }
+
+        private ClientAppRepo _clientAppRepo;
+        private ApiClient _testClient;
+
+
+        [TestFixture]
```

Comment:
I wrote a lot of tests in here, trying to cover many of the basic functions in `ClientAppRepo`, a few of the advanced functions, and especially the functions that were modified in branch `ODS-6250`.